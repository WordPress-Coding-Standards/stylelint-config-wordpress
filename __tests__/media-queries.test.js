'use strict';

const fs = require( 'fs' ),
	config = require( '../' ),
	stylelint = require( 'stylelint' ),
	validCss = fs.readFileSync( './__tests__/media-queries-valid.css', 'utf-8' ),
	invalidCss = fs.readFileSync( './__tests__/media-queries-invalid.css', 'utf-8' );

describe( 'flags no warnings with valid media queries css', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint({
			code: validCss,
			config,
		});
	});

	it( 'did not error', () => {
		return result.then( data => (
			expect( data.errored ).toBeFalsy()
		) );
	});

	it( 'flags no warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toHaveLength( 0 )
		) );
	});
});

describe( 'flags warnings with invalid media queries css', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint({
			code: invalidCss,
			config,
		});
	});

	it( 'did error', () => {
		return result.then( data => (
			expect( data.errored ).toBeTruthy()
		) );
	});

	it( 'flags correct number of warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toHaveLength( 11 )
		) );
	});

	it( 'snapshot matches warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toMatchSnapshot()
		) );
	});
});
