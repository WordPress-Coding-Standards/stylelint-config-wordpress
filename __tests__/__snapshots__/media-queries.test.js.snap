// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags warnings with invalid media queries css snapshot matches warnings 1`] = `
Array [
  Object {
    "column": 1,
    "line": 31,
    "rule": "at-rule-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown at-rule \\"@mdia\\" (at-rule-no-unknown)",
  },
  Object {
    "column": 26,
    "line": 1,
    "rule": "media-feature-colon-space-after",
    "severity": "error",
    "text": "Expected single space after \\":\\" (media-feature-colon-space-after)",
  },
  Object {
    "column": 27,
    "line": 6,
    "rule": "media-feature-colon-space-after",
    "severity": "error",
    "text": "Expected single space after \\":\\" (media-feature-colon-space-after)",
  },
  Object {
    "column": 27,
    "line": 6,
    "rule": "media-feature-colon-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\":\\" (media-feature-colon-space-before)",
  },
  Object {
    "column": 17,
    "line": 11,
    "rule": "media-feature-name-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown media feature name \\"max-width 699px\\" (media-feature-name-no-unknown)",
  },
  Object {
    "column": 28,
    "line": 16,
    "rule": "media-feature-range-operator-space-after",
    "severity": "error",
    "text": "Expected single space after range operator (media-feature-range-operator-space-after)",
  },
  Object {
    "column": 29,
    "line": 21,
    "rule": "media-feature-range-operator-space-after",
    "severity": "error",
    "text": "Expected single space after range operator (media-feature-range-operator-space-after)",
  },
  Object {
    "column": 25,
    "line": 16,
    "rule": "media-feature-range-operator-space-before",
    "severity": "error",
    "text": "Expected single space before range operator (media-feature-range-operator-space-before)",
  },
  Object {
    "column": 25,
    "line": 26,
    "rule": "media-feature-range-operator-space-before",
    "severity": "error",
    "text": "Expected single space before range operator (media-feature-range-operator-space-before)",
  },
  Object {
    "column": 27,
    "line": 36,
    "rule": "media-query-list-comma-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\",\\" (media-query-list-comma-space-before)",
  },
  Object {
    "column": 27,
    "line": 40,
    "rule": "media-query-list-comma-space-before",
    "severity": "error",
    "text": "Unexpected whitespace before \\",\\" (media-query-list-comma-space-before)",
  },
]
`;
